src/bans.rs

```diff
@@ -444,6 +445,27 @@ pub fn check(
             }
         }
 
+        if let Some(ref allow_build_scripts) = allow_build_scripts {
+            let has_build_script = krate
+                .targets
+                .iter()
+                .any(|t| t.kind.iter().any(|k| *k == "custom-build"));
+
+            let allowed_build_script = allow_build_scripts.value.iter().any(|id| {
+                krate.name == id.name && crate::match_req(&krate.version, id.version.as_ref())
+            });
+
```

Comment:
This should be moved inside `if has_build_script {` since it only needs to be checked in that case